Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Collections.Specialized;
+    using System.Globalization;
+    using System.Linq;
+
+    /// <summary>
+    /// WebHeaderCollection extension methods.
+    /// </summary>
+    internal static class WebHeaderCollectionExtensions
+    {
+        /// <summary>
+        /// For the given header collection, for a given header of name-value type, find the value of a particular key.
+        /// </summary>
+        /// <param name="headers">Header collection.</param>
+        /// <param name="headerName">Name of the header in the collection.</param>
+        /// <param name="keyName">Desired key of the key-value list.</param>
+        /// <returns>Value against the given parameters.</returns>
+        public static string GetNameValueHeaderValue(this NameValueCollection headers, string headerName, string keyName)
```

Comment:
Well I'm not using the NameValueHeaderValue methods - so can't comment on whether they are safe or not and lack the necessary historical context that you are mentioning.